service/src/main/java/org/cbioportal/service/impl/GeneServiceImpl.java

```diff
@@ -31,50 +31,138 @@
 */
 package org.cbioportal.service.impl;
 
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
 import org.cbioportal.model.Gene;
+import org.cbioportal.model.meta.BaseMeta;
 import org.cbioportal.persistence.GeneRepository;
 import org.cbioportal.service.GeneService;
+import org.cbioportal.service.exception.GeneNotFoundException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-/**
- *
- * @author jiaojiao
- */
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Service
-public class GeneServiceImpl implements GeneService{
+public class GeneServiceImpl implements GeneService {
+
     @Autowired
     private GeneRepository geneRepository;
-    
+
     @Override
-    public List<Gene> getGeneListByHugoSymbols(List<String> hugo_gene_symbol){
+    public List<Gene> getAllGenes(String projection, Integer pageSize, Integer pageNumber, String sortBy,
+                                  String direction) {
 
-        List<Gene> genes = geneRepository.getGeneListByHugoSymbols(hugo_gene_symbol);
-        for (Gene gene : genes) {
-            gene.setChromosome(getChromosome(gene));
+        List<Gene> geneList = geneRepository.getAllGenes(projection, pageSize, pageNumber, sortBy, direction);
+
+        for (Gene gene : geneList) {
+            gene.setChromosome(getChromosome(gene.getCytoband()));
         }
 
-        return genes;
+        return geneList;
     }
 
-    private String getChromosome(Gene gene) {
+    @Override
+    public BaseMeta getMetaGenes() {
 
-        if (gene.getCytoband() == null) {
+        return geneRepository.getMetaGenes();
+    }
+
+    @Override
+    public Gene getGene(String geneId) throws GeneNotFoundException {
+
+        Gene gene;
+
+        if (isInteger(geneId)) {
```

Comment:
logic for detecting Integer v. Hugo argument. Since this is argument parsing, it might have gone in the web layer, but it is fine here too I think.